### PR TITLE
Add a test for HTTP.request with AWS authentication

### PR DESCRIPTION
**Pull request by ararslan**
The test makes a request to httpbin which returns the sent headers, then inspects the headers for consistency. Exercises the `AWS4AuthLayer` code paths.

### CODE REVIEW

**Comment by ararslan** [comment]
Julia 1.0 failures seem to be https://github.com/JuliaWeb/MbedTLS.jl/issues/193.

File: test/aws4.jl [hunk 1]
```diff
@@ -200,5 +200,28 @@ const required_headers = ["Authorization", "host", "x-amz-date"]
...
-end
 
+    @testset "HTTP.request with AWS authentication" begin
+        resp = HTTP.request("GET",
+                            "https://httpbin.org/headers";
+                            aws_authorization=true,
+                            timestamp=DateTime(2015, 8, 30, 12, 36),
+                            aws_service="service",
+                            aws_region="us-east-1",
+                            # NOTE: These are the example credentials as specified in the AWS docs,
```
Comment (iamed2) [inline test/aws4.jl]: Link to documentation?

Comment (ararslan) [reply test/aws4.jl]: It's at the top of the file.

**Comment by ararslan** [comment]
Doc link is in https://github.com/JuliaWeb/HTTP.jl/blob/master/test/aws4.jl#L7-L8

**Pull request merged**